### PR TITLE
Conditions

### PR DESCRIPTION
**Pull request by pollockcl**
SUBMITTED BY: 

Major Changes
  * Added conditions and admin/conditions controller
  * user/visitor/admin CRUD functionality
  * some routing shenanagins

Applicable Bug Fixes
  *
  *
  *

Passing Tests? (y)

REVIEWED BY:

### CODE REVIEW

File: app/helpers/admin/conditions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module Admin::ConditionsHelper
```
Comment (pollockcl) [inline app/helpers/admin/conditions_helper.rb]: Haha, I'll delete this.

File: app/controllers/admin/conditions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      if @condition.save
+        flash[:success] = 'Condition successfully created.'
+        redirect_to @condition
+      else
+        flash.now[:danger] = 'Something went wrong. Please try again.'
+        render :new
+      end
+    end
+
+    def edit
```
Comment (ssciolist) [inline app/controllers/admin/conditions_controller.rb]: @iandouglas @AliSchlereth We were wondering whether we should omit empty methods like this or whether it's better to have them for ~rails magic~  purposes.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -9,9 +9,9 @@
   </head>
 
   <body>
-    <% flash.each do |message_type, message| %>
-      <div class="alert alert-<%= message_type %>"><%= message %></div>
```
Comment (ssciolist) [inline app/views/layouts/application.html.erb]: lol.

File: config/routes.rb [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
+
   root 'welcome#index'
   get '/login', to: 'sessions#new'
   post '/login', to: 'sessions#create'
 
   get '/dashboard', to: 'user#show'
 
-  resources :stations
   namespace :admin do
+    resources :conditions, exclude: %i[index show]
```
Comment (ssciolist) [inline config/routes.rb]: Let's pick one standard for our resource file.

File: spec/features/conditions/visitor_can_see_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  describe 'conditions index page' do
+    it 'they see all conditions' do
+      guest = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(guest)
+      condition1 = create(:con1)
+      condition2 = create(:con2)
+
+      visit conditions_path
+
+      condition1.attributes.each_value do |attr|
```
Comment (ssciolist) [inline spec/features/conditions/visitor_can_see_index_spec.rb]: I love this, thanks for showing us this.

File: app/controllers/admin/conditions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
+module Admin
+  class ConditionsController < BaseController
+    before_action :set_condition, only: %i[edit update destroy]
```
Comment (ssciolist) [inline app/controllers/admin/conditions_controller.rb]: I like this too, I'll have to remember to implement it for my admin controller.

File: app/controllers/admin/conditions_controller.rb (hunk 2 above)
Comment (AliSchlereth) [reply app/controllers/admin/conditions_controller.rb]: @pollockcl keep this for readability reasons.

**Pull request merged**